### PR TITLE
Add Pending State and State Message in Webapi Agent

### PR DESCRIPTION
**Pull request by Future-Outlier**
# TL;DR
As title.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [x] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
For Databricks agent service, we need to have a pending state.
Now we can see what's happening in the pending state!

image

## Follow-up issue
https://github.com/flyteorg/flyteidl/pull/440
https://github.com/flyteorg/flytekit/pull/1834

### CODE REVIEW

**Comment by Future-Outlier** [comment]
@pingsutw Just fixed my error, please review!
Thanks a lot!

**Comment by Future-Outlier** [comment]
Please approve it again, thanks a lot!

File: go/tasks/plugins/webapi/agent/plugin_test.go [hunk 1]
```diff
@@ -99,4 +100,106 @@ func TestPlugin(t *testing.T) {
...
 		assert.NotEqual(t, context.TODO(), ctx)
 	})
+
+	t.Run("test PENDING Status", func(t *testing.T) {
+		plugin := Plugin{
+			metricScope: fakeSetupContext.MetricsScope(),
+			cfg:         GetConfig(),
+		}
+		taskContext := new(webapiPlugin.StatusContext)
+
```
Comment (pingsutw) [inline go/tasks/plugins/webapi/agent/plugin_test.go:110]: Could we move this code to line 29, so we don't have to declare it again in each test?

Comment (Future-Outlier) [reply go/tasks/plugins/webapi/agent/plugin_test.go]: We have to declare "taskContext " in each test, but "plugin " is not needed, thanks a lot for review!

**Comment by Future-Outlier** [comment]
Please review again, thanks a lot!

File: go/tasks/plugins/webapi/agent/plugin.go [hunk 2]
```diff
@@ -162,6 +163,8 @@ func (p Plugin) Status(ctx context.Context, taskCtx webapi.StatusContext) (phase
 	taskInfo := &core.TaskInfo{}
 
 	switch resource.State {
+	case admin.State_PENDING:
+		return core.PhaseInfoQueued(time.Now(), core.DefaultPhaseVersion, "Job Created"), nil
```
Comment (pingsutw) [inline go/tasks/plugins/webapi/agent/plugin.go:167]: sorry, I think we should use PhaseInfoQueuedWithTaskInfo here. https://github.com/flyteorg/flyteplugins/blob/552f145c4632944a23e7e2d9fa2b2f065c4b14b9/go/tasks/plugins/webapi/bigquery/plugin.go#L287

**Review by pingsutw — APPROVED** [review]

File: go/tasks/plugins/webapi/agent/plugin.go (hunk 2 above)
Comment (Future-Outlier) [reply go/tasks/plugins/webapi/agent/plugin.go:167]: No problem!
Thanks a lot for spending time reviewing this!

**Comment by Future-Outlier** [comment]
Just tested it in Databricks agent service, please review it, thanks a lot!

**Comment by kumare3** [comment]
@pingsutw / @Future-Outlier can we add more states like initializing etc, all the ones we already have in the task phases?

Also have messages for each phase sent by the agent

File: go/tasks/plugins/webapi/agent/plugin.go [hunk 3]
```diff
@@ -162,6 +162,8 @@ func (p Plugin) Status(ctx context.Context, taskCtx webapi.StatusContext) (phase
 	taskInfo := &core.TaskInfo{}
 
 	switch resource.State {
+	case admin.State_PENDING:
+		return core.PhaseInfoQueuedWithTaskInfo(core.DefaultPhaseVersion, "Job is PENDING", taskInfo), nil
```
Comment (kumare3) [inline go/tasks/plugins/webapi/agent/plugin.go:166]: I don't like the message. This message does not convey any information to the user. Can the agent send this message

Comment (Future-Outlier) [reply go/tasks/plugins/webapi/agent/plugin.go:166]: @kumare3 
https://github.com/flyteorg/flyteplugins/blob/503d1be56eeca69b29606082905d8c3f3dbb86d7/go/tasks/pluginmachinery/core/phase.go#L220-L226
Is this solution suitable?

**Comment by eapolinario** [comment]
Hi, we are moving all Flyte development to a monorepo. In order to help the transition period, we're moving open PRs to monorepo automatically and your PR was moved to . Notice that if there are any conflicts in the resulting PR they most likely happen due to the change in the import path of the flyte components.

**Comment by eapolinario** [comment]
Hi, we are moving all Flyte development to a monorepo. In order to help the transition period, we're moving open PRs to monorepo automatically and your PR was moved to https://github.com/flyteorg/flyte/pull/4133. Notice that if there are any conflicts in the resulting PR they most likely happen due to the change in the import path of the flyte components.

**Pull request closed without merging**